src/schema/v2/lot.ts

```diff
@@ -95,41 +95,31 @@ const resolveWatchedLotConnection = async (_parent, args, context) => {
     variables: {
       ids: saleArtworkIds,
     },
-  }).then((res: FetchResponse) => res.json())
-
-  const { data: causalityData, errors: causalityErrors } = lotStatesResponse
-
-  // If the causality request failed for some reason, throw its errors.
-  if (causalityErrors) {
-    const errors = causalityErrors.reduce((acc, ce) => {
-      return acc + " " + ce["message"]
-    }, "From causality: ")
-    throw new GraphQLError(errors)
-  } else {
-    // make a map of lot state data
-    const lotDataMap = causalityData.lots.reduce((acc, lot) => {
-      acc[lot.internalID] = lot
+  })
+
+  // make a map of lot state data
+  const lotDataMap = causalityData.lots.reduce((acc, lot) => {
+    acc[lot.internalID] = lot
+    return acc
+  }, {})
+
+  // place lotData in context for retrieval at stitching stage
+  context.lotDataMap = lotDataMap
+
+  // create nodes from sale artworks
+  const availableLotDataIds = Object.keys(lotDataMap)
+  const nodes = watchedSaleArtworks.reduce((acc, saleArtwork) => {
+    if (!availableLotDataIds.find((id) => id === saleArtwork._id)) {
+      console.warn(`lot state for ${saleArtwork._id} not found - skipping`)
```

Comment:
If this log will remain in prod it would be helpful to add a breadcrumb back to the location:
```ts
console.warn('[metaphysics @ schema/v2/lot] Warning: lots state for ...) 
```

Reply:
I'll do this for now but if there is a better way to log/track things like this I'm interested to hear it. This condition should be relatively common in staging and never (as far as i know) in production.

Reply:
You can use this one, but i'd still keep a bread crumb - they're always useful: https://github.com/damassi/metaphysics/blob/master/src/lib/loggers.ts#L3